src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -1451,35 +1415,31 @@ public StringBuilder Insert(int index, char[]? value, int startIndex, int charCo
 
             if (charCount > 0)
```

Comment:
Seems better to remove this check, too.  Seems better to optimize for the != 0 case than to optimize for the == 0 case.